### DISCUSSION TITLE
Is this Benchmark Suite abandoned?

### DISCUSSION THREAD

**Discussion by fredericosantos** [cavalab/srbench#218] [General]
Hello, I have seen very little activity on this repository. 

Is anyone interested in passing on the torch?

Some improvements that would be nice to see/be discussed:
1. A `bring your own optimizer` solution for `grid_time`. Currently it uses GridSearchCV, which is terrible for some datasets that have 6 sample sizes.
2. Clear instructions on time limits: Is the 6h limit per seed? Per dataset?
3. A refactoring of which datasets enter the benchmark. As an example, the 'true' R^2 of many of the datasets is lower than 1; which means finding the 'human true' formula is not obvious.
4. A discussion on using fixed seeds;

**Comment by lacava** [comment]
Our last push to master was 12 days ago and there are multiple active PRs. So, to answer your title question, no. 

1. We did this in 2022 https://cavalab.org/srbench/competition-2022/
2. Depends which version of the experiment you're referring to.
3. That's why we have "black box" and "ground truth" experiments, and experiment with noise.
4. Feel free.

> Is anyone interested in passing on the torch?

The torch is already shared. You're welcome to contribute. See https://cavalab.org/srbench/contributing/#ground-rules; read 3 carefully.

**Reply by fredericosantos** [reply]
Thank you for the reply. I am very happy that this project has not been abandoned, I would love to participate in improving it. Particularly, to bring the hyper parameter optimization step that exists in `srbench_2025` to `master`--but improved so that anyone can submit their own HPO. 
My confusion was because I've been working with the `srbench_2025` branch and it has not received any updates, nor was it merged with `master`.

Does (1) mean the branch `srbench_2025` is not the official way to use `SRBench`? Are there plans to merge `srbench_2025` onto `master`? 

I ran all my experiments on `srbench_2025` after attending the talk at GECCO 2025, but I'm starting to think I cannot open a PR, since I am making use of the hyper parameter optimization step (and possibly the branch code has diverged at this point).

I could do what Operon does, and pack the entire HPO step into `.fit()`, but that would severely affect the reported training times. In case HPO is not possible, I will follow Operon's `.fit()` way and substantially increase training time, but I would like confirmation beforehand.

Would you be able to provide some guidance into which version (master or srbench_2025) is currently supported? I have very good results, so I would very much like to be able to create the correct PR.

**Reply by lacava** [reply]
sorry for the delay. we plan to keep srbench_2025 as it is to mirror the call for action paper, but we'd like to take lessons learned from that and also our competitions and integrate them into master. the big push this summer has been do move towards a docker image per method setup, so environments and integration tests are isolated while still sharing the eval harness. 

when it comes to whether to wrap HPO in a method or have everyone get optimized, we have done both types of experiments as you know and there isn't a best answer. some people do not want to do HPO and some do. because of this, in our last competition rounds (srbench++ ieee paper) we switched to making this part of the algorithm / participants design, rather than a requirement. in the neurips paper we did a method-specific hyperparameter tuning for black box problems then ran the best setting on the synthetic problems. for the "call for action" workshop paper at gecco we did a small HPO fixed config for all algorithms. 

i guess my advice is to follow the design that corresponds to the results you're comparing to. if you're comparing to the feather files from the neurips paper, use that experiment design; if you're comparing to the call for action paper (sounds like you are), use that one. 

for submitting an algorithm to the repo, we generally want that to go to the dev branch, which ends up in main. this version does not have hyperparameter range requirements, so it's up to you if you want fit() to run HPO or not, but this is the version people will generally use out of the box. for future experiments we'd have a discussion beforehand about whether to do HPO but we'd do something pre-defined across the methods.
